src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +40,109 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
+      m_new_axis_mask_set(std::move(new_axis_mask)),
+      m_shrink_axis_mask_set(std::move(shrink_axis_mask)) {}
+
+    Result infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        // align with intel_cpu::node::StridedSlice
+        static constexpr size_t DATA_ID = 0, BEGIN_ID = 1, END_ID = 2, STRIDE_ID = 3;
+        const VectorDims& shapeIn = input_shapes[DATA_ID].get();
+        const VectorDims& shapeBegin = input_shapes[BEGIN_ID].get();
+        if (data_dependency.at(BEGIN_ID)->getDesc().getPrecision() != Precision::I32 ||
+            data_dependency.at(END_ID)->getDesc().getPrecision() != Precision::I32 ||
+            data_dependency.at(STRIDE_ID)->getDesc().getPrecision() != Precision::I32) {
+            IE_THROW(Unexpected) << "The data type of begin/end/stride is NOT I32, which is unexpected!";
+        }
+        auto beginPtr = reinterpret_cast<int32_t *>(data_dependency.at(BEGIN_ID)->GetPtr());
+        auto endPtr = reinterpret_cast<int32_t *>(data_dependency.at(END_ID)->GetPtr());
+        auto stridePtr = reinterpret_cast<int32_t *>(data_dependency.at(STRIDE_ID)->GetPtr());
+
+        for (auto i = 0, new_idx = 0; i < shapeIn.size(); ++i) {
+            if (m_new_axis_mask_set.count(i)) {
+                // deal with new_axis_mask
+                m_outputShape[new_idx] = 1;
+                m_outputShape[new_idx+1] = shapeIn[i];
+                new_idx+=2;
+            } else if (!m_shrink_axis_mask_set.count(i)) {
+                // deal with begin_mask and end_mask
+                if ((i >= shapeBegin[0]) || (shapeIn[i] == 0)) {
+                    m_outputShape[new_idx] = shapeIn[i];
+                } else {
+                    auto begin = m_begin_mask_set.count(i) ? 0 : beginPtr[i];
+                    auto end  = m_end_mask_set.count(i) ? shapeIn[i] : endPtr[i];
+                    m_outputShape[new_idx] = ov::op::slice::get_sliced_value(shapeIn[i], begin, end, stridePtr[i]);
+                }
+                new_idx += 1;
+            }
+        }
+        return {{m_outputShape}, ShapeInferStatus::success};
+    }
+
+    port_mask_t get_port_mask() const override {
+        static constexpr IShapeInfer::port_mask_t port_mask = PortMask(1, 2, 3, 4);
+        return port_mask;
+    }
+
+private:
+    VectorDims m_outputShape;
+    const std::unordered_set<int64_t> m_begin_mask_set;
+    const std::unordered_set<int64_t> m_end_mask_set;
+    const std::unordered_set<int64_t> m_new_axis_mask_set;
+    const std::unordered_set<int64_t> m_shrink_axis_mask_set;
+};
+
+class StridedSliceShapeInferFactory : public ShapeInferFactory {
+public:
+    StridedSliceShapeInferFactory(const std::shared_ptr<ov::Node>& op)
+    : m_op(op) {}
+    ShapeInferPtr makeShapeInfer() const override {
+        static constexpr IShapeInfer::port_mask_t port_mask = PortMask(1, 2, 3, 4);
```

Comment:
It is more logical to keep this mask in one place:
```cpp
class StridedSliceShapeInfer {
...
public:
static constexpr IShapeInfer::port_mask_t port_mask = PortMask(1, 2, 3, 4);
...


port_mask_t get_port_mask() const override {
    return port_mask;
}
...
};
```
And access just as follows:
```cpp
return std::make_shared<NgraphShapeInfer>(make_shape_inference(m_op), StridedSliceShapeInfer::port_mask );
...
```

Reply:
@maxnick 
Thanks for your advice. Appy your suggestion to `keep it in one place`.
But I didn't define the `port_mask` in the `StridedSliceShapeInfer`. The reason is the `port_mask` is not only called by `StridedSlice` shape infer. If Slice` shape infer also access `port_mask` from `StridedSlice` shape infer it in the near future, it seems a bit strange. So I define it at the top as other nodes do.